### PR TITLE
 gh-1059 : Added health indicator for kafka messages listener containers created via kafka binder

### PR DESCRIPTION
**Pull request by tdanylchuk**
Before I jump into tests, please let me know whether the approach is acceptable for you guys @garyrussell @sobychacko

### CODE REVIEW

File: spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java [hunk 1]
```diff
@@ -1479,6 +1483,10 @@ public void setProducerConfigCustomizer(ProducerConfigCustomizer producerConfigC
 		this.producerConfigCustomizer = producerConfigCustomizer;
 	}
 
+	public List<AbstractMessageListenerContainer<?, ?>> getKafkaMessageListenerContainers() {
+		return kafkaMessageListenerContainers;
```
Comment (garyrussell) [inline spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java:1487]: We should return `Collections.unModifiableList()` here.

It should also be package private - we don't want end users using this API.

**Review by garyrussell — CHANGES_REQUESTED** [review]
LGTM with these changes.

File: spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java (hunk 1 above)
Comment (tdanylchuk) [reply spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java]: done

**Comment by tdanylchuk** [comment]
Hey @garyrussell @sobychacko, super kind reminder =)

**Comment by sobychacko** [comment]
@tdanylchuk Is the PR ready for final review and merge? I thought I saw the WIP label on the title, thus thinking that you were still working on it.

**Comment by tdanylchuk** [comment]
nope, i've removed it, you may review now =)

**Comment by sobychacko** [comment]
Thank you @tdanylchuk for the PR. It is now merged upstream.

**Comment by tdanylchuk** [comment]
Thanks @sobychacko ! Also wonder, is there any chance we can port this change into 3.0.x?

**Comment by sobychacko** [comment]
@tdanylchuk It is now backported to 3.0.x.

**Comment by tdanylchuk** [comment]
awesome, thanks again!

**Pull request closed without merging**